File: src/keep2shareSDK/keep2shareSDK/JSON.vb
Imports Newtonsoft.Json

Namespace JSON


    Public Class JSON_Error
         <JsonProperty("message", NullValueHandling:=NullValueHandling.Ignore)> Public Property ErrorMessage As String
    End Class

#Region "RequestCaptcha"
    Public Class JSON_RequestCaptcha
        <JsonProperty("challenge")> Public Property ChallengeID As String
        Private _captcha_url As String
        Public Property captcha_url As String
            Get
                Return Net.WebUtility.UrlDecode(_captcha_url)
            End Get
            Set(value As String)
                _captcha_url = value
            End Set
        End Property
        Public Property captcha_image As Drawing.Image
    End Class
#End Region

#Region "TestAccessToken"
    Public Class JSON_TestAccessToken
        Public Property status As String
        Public Property code As Integer
        Public ReadOnly Property Success As Boolean
            Get
                Return If(status = "success", True, False)
            End Get
        End Property
    End Class
#End Region

#Region "AccountInfo"
    Public Class JSON_AccountInfo
        Public Property available_traffic As Long
        Public Property account_expires As Boolean
    End Class
#End Region

#Region "FilesList"
    Public Class JSON_FilesList
        <JsonProperty("files")> Public Property FilesList As List(Of JSON_FileMetadata)
    End Class
    Public Class JSON_FileMetadata
        Public Property id As String
        Public Property name As String
        Public Property is_available As Boolean
        Public Property is_folder As Boolean
        Public Property date_created As String
        Public Property access As String
        Public Property size As String
        Public Property md5 As String
        Public Property extended_info As JSON_FileMetadataExtendedInfo
        Enum file_OR_folder
            File
            Folder
        End Enum
        Public ReadOnly Property FileORFolder As file_OR_folder
            Get
                Return If(is_folder, file_OR_folder.Folder, file_OR_folder.File)
            End Get
        End Property
    End Class
    Public Class JSON_FileMetadataExtendedInfo
        Public Property storage_object As String
        Public Property size As String
        Public Property date_download_last As String
        Public Property downloads As String
        Public Property access As String
        Public Property content_type As String
    End Class
#End Region

#Region "CreateNewFolder"
    Public Class JSON_CreateNewFolder
        Public Property id As String
    End Class
#End Region

#Region "RemoteUpload"
    Public Class JSON_RemoteUpload
        Public Property acceptedUrls As List(Of Acceptedurl)
        Public Property rejectedUrls As List(Of Rejectedurl)
    End Class
    Public Class Acceptedurl
        Public Property url As String
        <JsonProperty("id")> Public Property JobID As String
    End Class
    Public Class Rejectedurl
        Public Property url As String
        Public Property message As String
    End Class
#End Region

#Region "JSON_RemoteUploadStatus"
    Public Class JSON_RemoteUploadStatus
        Public Property status As String
        Public Property code As Integer
        Public ReadOnly Property Success As Boolean
            Get
                Return If(status = "success", True, False)
            End Get
        End Property
        <JsonExtensionData> Public Property uploads As Dictionary(Of String, Object)
    End Class

    Public Class Uploads
        Public Property _11522884 As _11522884
        Public Property _11522888 As _11522888
    End Class

    Public Class _11522884
        Public Property status As String
        Public Property progress As String
    End Class

    Public Class _11522888
        Public Property status As String
        Public Property progress As String
    End Class
#End Region

#Region "JSON_GetUploadUrl"
    Public Class JSON_GetUploadUrl
        Public Property form_action As String
        Public Property file_field As String
        Public Property form_data As JSON_GetUploadUrlFormData
    End Class
    Public Class JSON_GetUploadUrlFormData
        Public Property ajax As Boolean
        Public Property params As String
        Public Property signature As String
    End Class
#End Region

#Region "JSON_Upload"
    Public Class JSON_Upload
        Public Property status_code As Integer
        Public Property success As Boolean
        Public Property user_file_id As String
        Public Property link As String
    End Class
#End Region

#Region "JSON_ListFolders"
    Public Class JSON_ListFolders
        Public Property foldersList As List(Of String)
    End Class
#End Region


End Namespace

File: keep2shareSDK/keep2shareSDK/JSON.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace keep2shareSDK.JSON
{

    #region JSON_Error
    public class JSON_Error
    {
        [JsonProperty("message")]public string ErrorMessage { get; set; }
    }
    #endregion

    #region JSON_RequestCaptcha
    public class JSON_RequestCaptcha
    {
        [JsonProperty("challenge")]
        public string ChallengeID { get; set; }
        private string _captcha_url;
        public string captcha_url
        {
            get
            {
                return System.Net.WebUtility.UrlDecode(_captcha_url);
            }
            set
            {
                _captcha_url = value;
            }
        }
        public System.Drawing.Image captcha_image { get; set; }
    }
    #endregion

    #region JSON_TestAccessToken
    public class JSON_TestAccessToken
    {
        public string status { get; set; }
        public int code { get; set; }
        public bool Success
        {
            get
            {
                return status == "success" ? true : false;
            }
        }
    }
    #endregion

    #region JSON_AccountInfo
    public class JSON_AccountInfo
    {
        public long available_traffic { get; set; }
        public bool account_expires { get; set; }
    }
    #endregion

    #region JSON_FilesList
    public class JSON_FilesList
    {
        [JsonProperty("files")]
        public List<JSON_FileMetadata> FilesList { get; set; }
    }
    #endregion

    #region JSON_FileMetadata
    public class JSON_FileMetadata
    {
        public string id { get; set; }
        public string name { get; set; }
        public bool is_available { get; set; }
        public bool is_folder { get; set; }
        public string date_created { get; set; }
        public string access { get; set; }
        public string size { get; set; }
        public string md5 { get; set; }
        public JSON_FileMetadataExtendedInfo extended_info { get; set; }
      public   enum file_OR_folder
        {
            File,
            Folder
        }
        public file_OR_folder FileORFolder
        {
            get
            {
                return is_folder ? file_OR_folder.Folder : file_OR_folder.File;
            }
        }
    }
    public class JSON_FileMetadataExtendedInfo
    {
        public string storage_object { get; set; }
        public string size { get; set; }
        public string date_download_last { get; set; }
        public string downloads { get; set; }
        public string access { get; set; }
        public string content_type { get; set; }
    }
    #endregion

    #region JSON_CreateNewFolder
    public class JSON_CreateNewFolder
    {
        public string id { get; set; }
    }
    #endregion

    #region JSON_RemoteUpload
    public class JSON_RemoteUpload
    {
        public List<Acceptedurl> acceptedUrls { get; set; }
        public List<Rejectedurl> rejectedUrls { get; set; }
    }
    public class Acceptedurl
    {
        public string url { get; set; }
        [JsonProperty("id")]
        public string JobID { get; set; }
    }
    public class Rejectedurl
    {
        public string url { get; set; }
        public string message { get; set; }
    }
    #endregion

    #region JSON_RemoteUploadStatus
    public class JSON_RemoteUploadStatus
    {
        public string status { get; set; }
        public int code { get; set; }
        public bool Success
        {
            get
            {
                return status == "success" ? true : false;
            }
        }
        [JsonExtensionData]
        public Dictionary<string, object> uploads { get; set; }
    }
    #endregion

    #region Uploads
    public class Uploads
    {
        public _11522884 _11522884 { get; set; }
        public _11522888 _11522888 { get; set; }
    }

    public class _11522884
    {
        public string status { get; set; }
        public string progress { get; set; }
    }

    public class _11522888
    {
        public string status { get; set; }
        public string progress { get; set; }
    }
    #endregion

    #region JSON_GetUploadUrl
    public class JSON_GetUploadUrl
    {
        public string form_action { get; set; }
        public string file_field { get; set; }
        public JSON_GetUploadUrlFormData form_data { get; set; }
    }
    public class JSON_GetUploadUrlFormData
    {
        public bool ajax { get; set; }
        public string @params { get; set; }
        public string signature { get; set; }
    }

    #endregion

    #region JSON_Upload
    public class JSON_Upload
    {
        public int status_code { get; set; }
        public bool success { get; set; }
        public string user_file_id { get; set; }
        public string link { get; set; }
    }
    #endregion

    public class JSON_ListFolders
    {
        public List<string> foldersList { get; set; }
    }


}
